vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.cc

```diff
@@ -339,31 +361,95 @@ void Wallet::OnTransferAnonToExternalWalletBalance(
       &public_key,
       &new_secret_key);
   if (!success) {
+    return "";
+  }
+
+  std::string header_signature = braveledger_bat_helper::sign(
+      header_keys,
+      header_values,
+      1,
+      "primary",
+      new_secret_key);
+
+  base::Value payload(base::Value::Type::DICTIONARY);
+
+  base::Value signed_tx(base::Value::Type::DICTIONARY);
+  signed_tx.SetStringKey("octets", octets);
+
+  base::Value denomination(base::Value::Type::DICTIONARY);
+  denomination.SetStringKey("amount", user_funds);
+  denomination.SetStringKey("currency", "BAT");
+
+  base::Value body(base::Value::Type::DICTIONARY);
+  body.SetStringKey("destination", new_address);
+  body.SetKey("denomination", std::move(denomination));
+  signed_tx.SetKey("body", std::move(body));
+
+  base::Value headers(base::Value::Type::DICTIONARY);
+  headers.SetStringKey("digest", header_digest);
+  headers.SetStringKey("signature", header_signature);
+  signed_tx.SetKey("headers", std::move(headers));
+
+  payload.SetKey("signedTx", std::move(signed_tx));
+  payload.SetStringKey("anonymousAddress", anon_address);
+
+  std::string json;
+  base::JSONWriter::Write(payload, &json);
+
+  return json;
+}
+
+void Wallet::OnTransferAnonToExternalWalletAddress(
+    ledger::Result result,
+    const std::string& anon_address,
+    const std::string& new_address,
+    const std::string& user_funds,
+    ledger::TransferAnonToExternalWalletCallback callback) {
+  if ((result != ledger::Result::LEDGER_OK &&
+      result != ledger::Result::ALREADY_EXISTS)
+      || anon_address.empty()) {
     callback(ledger::Result::LEDGER_ERROR);
     return;
   }
 
-  std::string headerSignature = braveledger_bat_helper::sign(header_keys,
-                                                             header_values,
-                                                             1,
-                                                             "primary",
-                                                             new_secret_key);
+  if (result != ledger::Result::ALREADY_EXISTS) {
+    ledger_->SetStringState(ledger::kStateUpholdAnonAddress, anon_address);
+  }
+
+  const std::string path = base::StringPrintf(
+      "%s%s/claim",
```

Comment:
%s%s is this correct or should it be %s/%s

Reply:
this is correct